torch/csrc/jit/runtime/profiling_record.cpp

```diff
@@ -61,16 +64,30 @@ void ProfilingRecord::insertShapeProfile(Node* n, Value* i) {
   n->replaceInputWith(i, pn->output());
 }
 
+bool isProfilable(Node* n) {
+
+  if (IsFusibleOperation(n) || isSupported(n)) {
+    return true;
+  }
+
+  switch (n->kind()) {
+    case prim::Return:
+    case prim::Param:
+    case prim::GetAttr:
+    case aten::t:
+      return true;
+    default:
+    return false;
+  }
+}
+
 void ProfilingRecord::instrumentBlock(Block* block) {
   for (auto it = block->nodes().begin(); it != block->nodes().end(); ++it) {
     auto n = *it;
     for (auto i : n->inputs()) {
-      if (!i->type()->isSubtypeOf(TensorType::get()) ||
-          i->node()->kind() == prim::profile) {
-        continue;
+      if (i->type()->isSubtypeOf(TensorType::get()) && (isProfilable(n) || isProfilable(i->node()))) {
```

Comment:
Not necessary to address in this PR, but I believe the logic should be "do we need to profile this Value?" rather than "do we need to profile this Value->Node edge?" since the profiling nodes would yield the same for all Value->Node edges coming from the same Value.

Reply:
in a general case, it might not if there was a mutating operation between two uses, so we indeed profiling edges or uses.